spec/create_new_milestone_action_spec.rb

```diff
@@ -0,0 +1,160 @@
+
+      before do
+        allow(Fastlane::Helper::GithubHelper).to receive(:new).and_return(github_helper)
+      end
+
+      it 'uses default value when neither `GHHELPER_NUMBER_OF_DAYS_FROM_CODE_FREEZE_TO_RELEASE` environment variable nor parameter `:number_of_days_from_code_freeze_to_release` is present' do
+        default_code_freeze_days = 14
+        expect(github_helper).to receive(:create_milestone).with(
+          anything,
+          anything,
+          anything,
+          anything,
+          default_code_freeze_days,
+          anything
+        )
+        run_described_fastlane_action(default_params)
+      end
+
+      it 'uses default value when neither `GHHELPER_MILESTONE_DURATION` environment variable nor parameter `:milestone_duration` is present' do
+        default_milestone_duration = 14
+        expect(github_helper).to receive(:create_milestone).with(
+          anything,
+          anything,
+          anything,
+          default_milestone_duration,
+          anything,
+          anything
+        )
+        run_described_fastlane_action(default_params)
+      end
+    end
+  end
+
+  describe 'calling the action validates input' do
+    it 'raises an error if no `GITHUB_TOKEN` environment variable nor parameter `:github_token` is present' do
+      ENV['GITHUB_TOKEN'] = nil
+      expect { run_action_without_key(:github_token) }.to raise_error(FastlaneCore::Interface::FastlaneError, "No value found for 'github_token'")
+    end
+
+    it 'raises an error if no `GHHELPER_REPOSITORY` environment variable nor parameter `:repository` is present' do
+      expect { run_action_without_key(:repository) }.to raise_error(FastlaneCore::Interface::FastlaneError, "No value found for 'repository'")
+    end
+
+    it 'raises an error if `need_appstore_submission:` parameter is passed as String' do
+      expect { run_action_with(:need_appstore_submission, 'foo') }.to raise_error "'need_appstore_submission' value must be either `true` or `false`! Found String instead."
+    end
+
+    it 'raises an error if `milestone_duration:` parameter is passed as String' do
+      expect { run_action_with(:milestone_duration, 'foo') }.to raise_error "'milestone_duration' value must be a Integer! Found String instead."
+    end
+
+    it 'raises an error if `number_of_days_from_code_freeze_to_release:` parameter is passed as String' do
+      expect { run_action_with(:number_of_days_from_code_freeze_to_release, 'foo') }.to raise_error "'number_of_days_from_code_freeze_to_release' value must be a Integer! Found String instead."
+    end
+  end
+
+  def run_action_without_key(key)
+    run_described_fastlane_action(default_params.except(key))
+  end
+
+  def run_action_with(key, value)
+    values = default_params
+    values[key] = value
+    run_described_fastlane_action(values)
+  end
+
+  def default_params
+    {
+      repository: test_repository,
+      need_appstore_submission: false,
+      github_token: test_token
+    }
+  end
```

Comment:
Same suggestions as for the other spec — `run_action_without(key:)`, `run_action_with(**additional_keys_and_values)`, and `let(:default_params)`

Reply:
I wonder if we can DRY those too using the [shared_context](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-context), or if this will be too overkill, as we also need to have the `default_params` defined on another place, to share the operations with those two methods 🤔